services/processor/native/adapter/fake/fake_compiler.go

```diff
@@ -31,18 +32,18 @@ func NewCompiler() *fakeCompiler {
 	}
 }
 
-func (c *fakeCompiler) ProvideFakeContract(fakeContractInfo *sdkContext.ContractInfo, code string) {
+func (c *fakeCompiler) ProvideFakeContract(fakeContractInfo *sdkContext.ContractInfo, code ...string) {
 	c.mutex.Lock()
 	defer c.mutex.Unlock()
 
-	c.provided[code] = fakeContractInfo
+	c.provided[code[0]] = fakeContractInfo
 }
 
-func (c *fakeCompiler) Compile(ctx context.Context, code string) (*sdkContext.ContractInfo, error) {
+func (c *fakeCompiler) Compile(ctx context.Context, code ...string) (*sdkContext.ContractInfo, error) {
 	c.mutex.RLock()
 	defer c.mutex.RUnlock()
 
-	contractInfo, found := c.provided[code]
+	contractInfo, found := c.provided[code[0]]
```

Comment:
this may cause weird issues in tests, if someone attempts to fake compilation, for some odd reason, of a contract with more than one file. Please either make the fake support multiple files properly, or panic if attempting to fake a contract with more than a single file